src/battle/battle_controller_player.c

```diff
@@ -1006,3 +1008,696 @@ void BattleControllerPlayer_UpdateFieldCondition(BattleSystem *bsys, BATTLECONTE
+       ctx->moveNoTemp = ctx->battleMons[ctx->battlerIdAttacker].unk88.encoredMove;
+       flag = 1;
+   } else if (ctx->battleMons[ctx->battlerIdAttacker].unk88.encoredMove &&
+               ctx->battleMons[ctx->battlerIdAttacker].unk88.encoredMove != ctx->battleMons[ctx->battlerIdAttacker].moves[ctx->battleMons[ctx->battlerIdAttacker].unk88.encoredMoveIndex]) {
+       ctx->moveNoTemp = ctx->battleMons[ctx->battlerIdAttacker].moves[ctx->battleMons[ctx->battlerIdAttacker].unk88.encoredMoveIndex];
+       ctx->battleMons[ctx->battlerIdAttacker].unk88.encoredMove = 0;
+       ctx->battleMons[ctx->battlerIdAttacker].unk88.encoredMoveIndex = 0;
+       ctx->battleMons[ctx->battlerIdAttacker].unk88.encoredTurns = 0;
+       flag = 1;
+   } else if (!Battler_CanSelectAction(ctx, ctx->battlerIdAttacker)) {
+       ctx->moveNoTemp = ctx->moveNoLockedInto[ctx->battlerIdAttacker];
+   } else if (ctx->unk_30B4[ctx->battlerIdAttacker] != ctx->battleMons[ctx->battlerIdAttacker].moves[ctx->movePos[ctx->battlerIdAttacker]]) {
+       ctx->moveNoTemp = ctx->battleMons[ctx->battlerIdAttacker].moves[ctx->movePos[ctx->battlerIdAttacker]];
+       flag = 1;
+   } else {
+       ctx->moveNoTemp = ctx->battleMons[ctx->battlerIdAttacker].moves[ctx->movePos[ctx->battlerIdAttacker]];
+   }
+   ctx->moveNoCur = ctx->moveNoTemp;
+   ctx->command = CONTROLLER_COMMAND_23;
+   ctx->battlerIdTarget = ov12_022506D4(bsys, ctx, ctx->battlerIdAttacker, ctx->moveNoTemp, flag, 0);
+   ov12_022642F0(bsys);
+}
+
+//static
+void BattleControllerPlayer_ItemInput(BattleSystem *bsys, BATTLECONTEXT *ctx) {
+    BattleItem *item;
+    int script;
+    
+    ctx->battlerIdAttacker = ctx->unk_21E8[ctx->unk_EC];
+    ctx->battlerIdTarget = ov12_02253DA0(bsys, ctx, ctx->battlerIdAttacker);
+    item = (BattleItem *)&ctx->unk_21A8[ctx->battlerIdAttacker][2];
+    
+    if (BattleSystem_GetFieldSide(bsys, ctx->battlerIdAttacker)) {
+        switch (ctx->unk_334.unk9D[ctx->battlerIdAttacker >> 1]) {
+        case 0:
+            script = 289;
+            break;
+        case 1:
+            script = 267;
+            break;
+        case 2:
+            if ((ctx->unk_334.unk9F[ctx->battlerIdAttacker >> 1] & 1) && (ctx->unk_334.unk9F[ctx->battlerIdAttacker >> 1] & 0x3e)) {
+                ctx->msgTemp = 6;
+            } else {
+                ctx->msgTemp = LowestFlagNo(ctx->unk_334.unk9F[ctx->battlerIdAttacker >> 1]);
+            }
+            script = 268;
+            break;
+        case 3:
+            ctx->msgTemp = ctx->unk_334.unk9F[ctx->battlerIdAttacker >> 1];
+            script = 269;
+            break;
+        case 4:
+            script = 270;
+            break;
+        }
+        ctx->itemTemp = ctx->unk_334.unkA0[ctx->battlerIdAttacker >> 1];
+    } else {
+        switch (item->page) {
+        case 0:
```

Comment:
do we have item page constants?

Reply:
not to my knowledge but I can double check and see if I can add them tomorrow